numpy/core/numeric.py

```diff
@@ -1509,11 +1509,14 @@ def normalize_axis_tuple(axis, ndim, argname=None, allow_duplicate=False):
     --------
     normalize_axis_index : normalizing a single scalar axis
     """
-    try:
-        axis = [operator.index(axis)]
-    except TypeError:
-        axis = tuple(axis)
-    axis = tuple(normalize_axis_index(ax, ndim, argname) for ax in axis)
+    # Speed-up most common cases.
+    if not isinstance(axis, (list, tuple)):
```

Comment:
Would be more correct to check `type(axis) in (list, tuple)`, since subclasses might override `__index__`.

I was tempted to change it to just `type(axis) is tuple`, to encourage people to use tuple rather than list axis arguments....

Reply:
I think if someone subclasses `list` or `tuple`, one should reasonably expect that the object can be iterated over, not also treated as an integer! (I'd actually prefer to just test for something being iterable, but things like `np.iterable` are *way* slower than the `isinstance` check.)

As for `tuple` - I can at least change the order!

Reply:
Reasonable or not, I'd prefer that we don't introduce an unnecessary behavior change now.

`type(axis) in (tuple, list)` sounds like a good approach to me